CRM/Core/Payment/Form.php

```diff
@@ -412,4 +412,39 @@ public static function getPaymentLabel($processor) {
 
   }
 
+
+  /**
+   * Adds in either a set of radio buttons or hidden fields to contain the payment processors on a front end form
+   *
+   * @param array $pamymentProcessors
+   * @param \CRM_Core_Form $form
+   * @param array $paymentProcessorObjects
+   * @param string|null $formType
+   */
+  public static function addFrontEndPaymentProcessors($pamymentProcessors, $form, $paymentProcessorObjects, $formType = NULL) {
+    $optAttributes = [];
+    foreach ($pamymentProcessors as $ppKey => $ppval) {
+      if ($ppKey > 0) {
+        $optAttributes[$ppKey]['class'] = 'payment_processor_' . strtolower($paymentProcessorObjects[$ppKey]['payment_processor_type']);
+      }
+      else {
+        $optAttributes[$ppKey]['class'] = 'payment_processor_paylater';
+      }
+    }
+    if (count($pamymentProcessors) > 1) {
+      $form->addRadio('payment_processor_id', ts('Payment Method'), $pamymentProcessors,
+        NULL, "&nbsp;", FALSE, $optAttributes
+      );
+    }
+    elseif (!empty($pamymentProcessors)) {
+      $ppKeys = array_keys($pamymentProcessors);
+      $currentPP = array_pop($ppKeys);
+      $form->addElement('hidden', 'payment_processor_id', $currentPP);
+      if ($currentPP === 0 && $formType === 'Contribute') {
```

Comment:
Let's not do this - the Contribute form can do this part itself

Reply:
ok moved it back now @eileenmcnaughton